CMakeLists.txt

```diff
@@ -1,3 +1,6 @@
+IF(NOT KOKKOS_HAS_TRILINOS)
+  cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+ENDIF()
```

Comment:
This won't work.  The variable `KOKKOS_HAS_TRILINOS` is not set at that point.
https://github.com/kokkos/kokkos/blob/2c2a8130d0ddabac4fe1649bc1a8c9a3364853b5/CMakeLists.txt#L34-L39